### PR TITLE
chore(userspace/libsinsp): keep a reference on m_sinsp_stats_v2 where needed

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

This PR keeps a reference on m_sinsp_stats_v2 where needed, instead of fetching it every time.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
chore!(userspace/libsinsp): keep a reference on m_sinsp_stats_v2 where needed 
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Mmmh seems like the job is hit by: https://github.com/actions/runner/issues/2402

File: userspace/libsinsp/metrics_collector.cpp
Comment (incertum) [inline userspace/libsinsp/metrics_collector.cpp:491]: Below we also have other metrics categories that then could not be used without the sinsip stats?

File: userspace/libsinsp/sinsp.cpp
Comment (incertum) [inline userspace/libsinsp/sinsp.cpp:1975]: I don't see that we init the values elsewhere now. We should keep that to avoid bogus numbers.

Comment (FedeDP) [reply userspace/libsinsp/sinsp.cpp:1975]: Nope, becase we use `make_shared` that will 0 initialize everything (check eg: https://stackoverflow.com/questions/45366323/does-make-shared-do-a-default-initialization-zero-init-for-each-member-variabl)
Also, multiple tests would fail it that wasn't the case!

File: userspace/libsinsp/metrics_collector.cpp
Comment (FedeDP) [reply userspace/libsinsp/metrics_collector.cpp:491]: I think this early return is ok, it has the exact same behavior as before as far as i can tell; otherwise, can you point me to the lines (perhaps i am just missing them!)

Comment (incertum) [reply userspace/libsinsp/metrics_collector.cpp:491]: https://github.com/falcosecurity/libs/blob/570b86c1d72783b888ff4c3131159fbb1bc628c7/userspace/libsinsp/metrics_collector.cpp#L724-L785

File: userspace/libsinsp/utils.cpp
Comment (jasondellaluce) [inline userspace/libsinsp/utils.cpp:971]: Care to make this thread-local or atomic?

File: userspace/libsinsp/sinsp.cpp
Comment (incertum) [reply userspace/libsinsp/sinsp.cpp:1975]: Thanks!

File: userspace/libsinsp/metrics_collector.cpp
Comment (FedeDP) [reply userspace/libsinsp/metrics_collector.cpp]: Fixed :)

**Comment by FedeDP** [comment]
> Mmmh seems like the job is hit by: https://github.com/actions/runner/issues/2402

So, the job is failing on our cncf x64 self-hosted ode too; but i manually tried on the node and it worked fine.
I think there is something wrong with how github actions spawn sub-process/shells (?)...

**Comment by FedeDP** [comment]
Will rebase on #1918 once it gets merged.

**Comment by FedeDP** [comment]
Rebased :) Wait for CI :crossed_fingers:

**Review by Andreagit97 — DISMISSED** [review]
/approve

**Comment by FedeDP** [comment]
>      4.27%     +9.52%  [.] sinsp_filter_check::tostring

Weird :/ perhaps latest changes (`thread_local`) killed the perf? @jasondellaluce

**Comment by FedeDP** [comment]
> Perf diff from master - scap file

Yes, scap files perf are pretty unstable it seems :(

**Comment by incertum** [comment]
Beautiful @FedeDP!

**Review by leogr — DISMISSED** [review]

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**
